bydbctl/internal/cmd/stream.go

```
@@ -0,0 +1,322 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/apache/skywalking-banyandb/pkg/version"
+	"github.com/ghodss/yaml"
+	"github.com/go-resty/resty/v2"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func newStreamCmd() *cobra.Command {
+	StreamCmd := &cobra.Command{
+		Use:     "stream",
+		Version: version.Build(),
+		Short:   "banyandb stream schema related Operation",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			return cmd.Parent().PersistentPreRunE(cmd.Parent(), args)
+		},
+	}
+
+	StreamCreateCmd := &cobra.Command{
+		Use:     "create",
+		Version: version.Build(),
+		Short:   "banyandb stream schema create Operation",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			return cmd.Parent().PersistentPreRunE(cmd.Parent(), args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			client := resty.New()
+			addr, err := cmd.Flags().GetString("addr")
+			if err != nil {
+				return err
+			}
+			body, err := cmd.Flags().GetString("json")
+			if err != nil {
+				return err
+			}
+			var data map[string]interface{}
+			err = json.Unmarshal([]byte(body), &data)
+			if err != nil {
+				return err
+			}
+			stream, ok := data["stream"].(map[string]interface{})
+			if !ok {
+				return errors.New("input json format error")
+			}
+			metadata, ok := stream["metadata"].(map[string]interface{})
+			if !ok {
+				return errors.New("input json format error")
+			}
+			_, ok = metadata["group"].(string)
+			if !ok {
+				metadata["group"] = fmt.Sprintf("%v", viper.Get("group"))
```

Could you return an error in case the group is absent?